File: ruby/EnemyStarShip.rb
module Deepspace

#Representa una nave enemiga
	class EnemyStarShip

		attr_reader :name, :ammoPower, :shieldPower, :loot, :damage

		def initialize(n,a,s,l,d)
			@name = n
			@ammoPower = a
			@shieldPower = s
			@loot = l
			@damage = d
		end

		def self.newCopy(e)
			new(e.name,e.ammoPower,e.shieldPower,e.loot,e.damage)
		end

		def getUIversion
			EnemyToUI.new(self)
		end

		def protection 
			shieldPower
		end

		def fire
			ammoPower
		end

		# Devuelve el resultado que se produce al recibir un disparo de una determinada potencia
		def receiveShot(shot)	
			if shieldPower < shot
				return ShotResult::DONOTRESIST
			else
				return ShotResult::RESIST
			end
		end

		def to_s
			"EnemyStarShip:\n	name: #{name}\n ammoPower: #{ammoPower}\n shieldPower: #{shieldPower}\n " +
			"loot: #{loot}\n damage: #{damage}\n"
		end

	end # class EnemyStarShip

end #module Deepspace

File: java/deepspace/src/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;


/**
 *
 * @author usuario
 */
public class EnemyStarShip implements SpaceFighter {
    
    private float ammoPower;
    private String name;
    private float shieldPower;
    private Loot loot;
    private Damage damage;
    
    EnemyStarShip(String n,float a,float s,Loot l,Damage d) {
        name = n;
        ammoPower = a;
        shieldPower = s;
        loot = l;
        damage = d;
    }
    
    EnemyStarShip(EnemyStarShip e) {
        name = e.name;
        ammoPower = e.ammoPower;
        shieldPower = e.shieldPower;
        loot = e.loot;
        damage = e.damage;
    }
    
    EnemyToUI getUIversion() {
        return new EnemyToUI(this);
    }
    
    public float fire() {
       return ammoPower; 
    }

    public float getAmmoPower() {
        return ammoPower;
    }

    public String getName() {
        return name;
    }

    public float getShieldPower() {
        return shieldPower;
    }

    public Loot getLoot() {
        return loot;
    }

    public Damage getDamage() {
        return damage;
    }
    
    public float protection() {
        return shieldPower;
    }
    
    public ShotResult receiveShot(float shot) {
        if(shieldPower < shot) {
            return ShotResult.DONOTRESIST;
        }
        else {
            return ShotResult.RESIST;
        }
    }
    
    public String toString(){
	
	String representacion = "EnemyStartShip \n"+
	    "ammoPower: " + ammoPower +
	    "\nname: " + name + 
	    "\nshieldPower :" + shieldPower;
	
	if (damage != null)
	    representacion += "\nDamage :" + damage.toString();
	
	representacion += "\n"; 

	return representacion; 
    }
    
}
